lib/LedgerSMB/Scripts/payment.pm

```diff
@@ -1115,8 +1115,11 @@ if (!$request->{exrate}) {
 # LETS GET THE DEPARTMENT INFO
 # WE HAVE TO SET $dbPayment->{department_id} in order to process
 if ($request->{department}) {
- $request->{department} =~ /^(\d+)--*/;
- $Payment->{department_id} = $1;
+    if ( $request->{department} =~ /^(\d+)--*/ ) {
+        $Payment->{department_id} = $1;
+    } else {
+        die "Error: Invalid data";
```

Comment:
It is a logic change.  Die as assertion.   The code implies the  author did not expect this match to ever fail, but this code does not control the input.

Assigning a usable default for the user seemed like bad business logic to me.

If ``department_id''s are constrained somewhere, please tell me where.